exonum/src/node/mod.rs

```diff
@@ -1075,23 +1089,12 @@ impl Node {
     }
 
     fn into_reactor(self) -> (HandlerPart<impl EventHandler>, NetworkPart, InternalPart) {
-        // TODO refactor node module to reduce complexity.
-        struct HandlerWithRuntime {
-            inner: NodeHandler,
-            system_runtime: SystemRuntime,
-        }
-
-        impl EventHandler for HandlerWithRuntime {
-            fn handle_event(&mut self, event: Event) {
-                match event {
-                    Event::Internal(InternalEvent::RestartApi) => self.system_runtime.restart(),
-                    event => self.inner.handle_event(event),
-                }
-            }
-        }
-
         let connect_message = self.state().our_connect_message().clone();
         let connect_list = self.state().connect_list().clone();
+
+        self.api_runtime_config
```

Comment:
Restarting HTTP servers should be tested. I've run into some problems with server restarts on my laptop with Windows 10; not sure whether this is Windows problem or something more generic.

Reply:
We should also consider splitting "Core API" and "Service API" into several independent entities. I've [somewhat tested](https://github.com/popzxc/exonum-python-backend/blob/master/python_runtime/exonum_runtime/runtime/runtime.py#L113) this approach in the Python runtime MVP and it seems to work just fine.

With that approach we will not have to restart HTTP server on each instance added.